src/test/java/test/DynamicGraphTest.java

```diff
@@ -74,15 +73,19 @@ public void test8() {
     dg.addEdge(0, c1, x, y);
     assertFreeNodesEquals(dg, y, x);
     dg.setStatus( dg.getFreeNodes(), Status.RUNNING);
-
-    assertFreeNodesEquals(dg, a1, a2);
+    assertFreeNodesEquals(dg);
+    
+    dg.setStatus(x, Status.FINISHED);
+    dg.setStatus(y, Status.FINISHED);
+    assertFreeNodesEquals(dg,a1);
+    
     dg.setStatus(dg.getFreeNodes(), Status.RUNNING);
 
     dg.setStatus(a1, Status.FINISHED);
     assertFreeNodesEquals(dg);
 
     dg.setStatus(a2, Status.FINISHED);
-    assertFreeNodesEquals(dg, b1, b2);
+    assertFreeNodesEquals(dg, b1);
 
     dg.setStatus(b2, Status.RUNNING);
```

Comment:
Correct. I don't believe the setups for these tests are valid. They have circular dependencies. I updated the tests such that if they were to be run, these are the free nodes.